### DISCUSSION TITLE
How do I make it generate compile_commands.json in vscode?

### DISCUSSION THREAD

**Discussion by DiogoM-Almeida** [friendlyanon/cmake-init#103] [Q&A]
When I generate the CMakeLists.txt manually and use CMake Tools I can generate it, but with the CMakeLists.txt here I am not able to generate it

**Comment by friendlyanon** [comment]
You have to make sure you are using a compatible generator and enable the [CMAKE_EXPORT_COMPILE_COMMANDS](https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html) variable in your configure preset. This is already done for you in the Linux and MacOS presets: https://github.com/friendlyanon/cmake-init/blob/master/cmake-init/templates/common/CMakeUserPresets.json#L23

On Windows, you'll have to switch to using Ninja as your generator. It's bundled with Visual Studio build tools along with CMake, so it'll be available from a [vcvars enabled command line](https://learn.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=msvc-170).

**Reply by DiogoM-Almeida** [reply]
I tried using CMAKE_EXPORT_COMPILE_COMMANDS in CMakeLists.txt , in CMakePresets.json and in CMakeUserPresets.json, none of them generated compile_commands.json.

I'm on Windows and I have ninja installed, does cmake-init use it by default?

I have msvc and clang as a compiler installed on my computer, when I generate CMakeLists.txt manually and use the CMake Tools extension, it asks which compiler I want to use. With cmake-init it asks for the preset, if so would I have to change something there?

**Reply by friendlyanon** [reply]
The generated project uses the VS2022 generator by default on Windows:  
https://github.com/friendlyanon/cmake-init/blob/master/cmake-init/templates/common/CMakePresets.json#L132  
https://github.com/friendlyanon/cmake-init/blob/master/cmake-init/templates/common/CMakeUserPresets.json#L35

**Reply by DiogoM-Almeida** [reply]
And is it possible to generate compile_commands.json with it?

**Reply by friendlyanon** [reply]
Please read the documentation:  
https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html

> **Note:** This option is implemented only by [Makefile Generators](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#makefile-generators) and [Ninja Generators](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#ninja-generators). It is ignored on other generators.

**Reply by DiogoM-Almeida** [reply]
sorry, but I really don't understand
I already tried adding this
I'm new to cmake, I don't know what I could be doing wrong
have also edited CMakePresets.json to use ninja, but when I do this it gives an error
